src/index-widget/index-widget.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  Component,
+  Input,
+  Inject,
+  forwardRef,
+  Optional,
+  SkipSelf,
+} from '@angular/core';
+import { Widget, BaseWidget } from '../base-widget';
+import {
+  InstantSearchInstance,
+  NgAisInstantSearch,
+} from '../instantsearch/instantsearch';
+import indexWidget from 'instantsearch.js/es/widgets/index/index';
+
+const connectIndex = () => indexWidget;
+
+@Component({
+  selector: 'ais-index',
+  template: `<ng-content></ng-content>`,
+})
+export class NgAisIndex extends BaseWidget {
+  @Input() public indexName: string;
+  @Input() public indexId?: string;
+
+  public widget?: Widget &
+    Pick<InstantSearchInstance, 'addWidgets' | 'removeWidgets'>;
+
+  constructor(
+    @SkipSelf()
```

Comment:
this is only necessary for this widget, because otherwise it tries to inject into the class itself, which throws an error

Reply:
Can you add this explanation as a comment since none of us are too familiar with Angular?